src/AutoRest.CSharp/Mgmt/AutoRest/MgmtOutputLibrary.cs

```diff
@@ -547,8 +548,8 @@ public bool TryGetRestClients(string requestPath, [MaybeNullWhen(false)] out Has
                     var originalResourcePath = operationSet.GetRequestPath();
                     var operations = GetChildOperations(originalResourcePath);
                     var resourceData = GetResourceData(originalResourcePath);
-                    if (resourceData is EmptyResourceData emptyResourceData)
-                        BuildPartialResource(requestPathToResources, resourceDataSchemaName, operationSet, operations, originalResourcePath, emptyResourceData);
+                    if (resourceData == OutputResourceData.Empty)
+                        BuildPartialResource(requestPathToResources, resourceDataSchemaName, operationSet, operations, originalResourcePath, resourceData);
```

Comment:
if all our partial resource are using the same instance of empty resource data as a place holder, maybe we do not need to pass it around.

Reply:
sounds good, updated